docs/source/compression/compression_evaluator.rst

```diff
@@ -0,0 +1,231 @@
+Compression Evaluator Creation
+==============================
+
+The evaluator is used to package the training and evaluation process for models with similar topologies.
+To explain why NNI needs an evaluator, let's first look at the general process of model compression in NNI.
+
+In model pruning, some algorithms need to prune according to some intermediate variables (gradients, activations, etc.) generated during the training process,
+and some algorithms need to gradually increase or adjust the sparsity of different layers during the training process,
+or adjust the pruning strategy according to the performance changes of the model during the pruning process.
+
+In model quantization, NNI has quantization-aware training algorithm,
+it can adjust the scale and zero point required for model quantization from time to time during the training process,
+and may achieve a better performance compare to post-training quantization.
+
+For the support of the above algorithms, NNI previously provided APIs like ``trainer``, ``traced_optimizer``, ``criterion``, ``finetuner``.
+These APIs were maybe tedious in terms of user experience. Users need to exchange the corresponding API frequently if they want to switch compression algorithms.
+After using the evaluator, users only need to consider one parameter to support NNI to complete the compression process.
+
+For users of PytorchLightning, the evaluator can be created with only a few lines of code, and there is no need to make changes to the original code in most cases.
+For users of native Pytorch, the evaluator requires the user to encapsulate the training process as a function and specifies the exposed interface,
+which will bring some complexity. And don't worry, in most cases, this will not change too much code.
+
+Here we give two examples of how to create an evaluator for both native Pytorch and PytorchLightning users.
+
+TorchEvaluator
+--------------
+
+``TorchEvaluator`` is for the users who work in a native Pytorch environment.
+Let's look at its API ``TorchEvaluator(training_func, optimizers, criterion, lr_schedulers, dummy_input, evaluating_func)``.
+
+training_func
+^^^^^^^^^^^^^
+Here is an example of ``training_func``.
+
+Training function has three required parameters, ``model``, ``optimizers`` and ``criterion``,
```

Comment:
This looks like an API reference.

A tutorial needs to give a concrete example for a concrete "end-to-end" use scenario.

Reply:
References are designed to be complicated.

If you intend to keep the complex part of reference in the "user guide", it's better to have a link to this user guide from the reference.

Reply:
As @QuanluZhang also considered that it should be placed in reference, 2>1, move these to docstring.

Also, good suggestion, add links to this user guide in reference.